### PR TITLE
feat: incoming offers page

### LINKED ISSUE #3450: Screen where creator can see incoming offers on Basilisk
We are missing one single screen where users can check their incoming offers on their NFTs

Show in table columns like

| Item | Amount | Offer Maker | Time | 
|--|--|--|--|
|nft name |bsx amount | maker address | relative date |

Route could be `/myoffers`

> Could user get some kind of notification of incoming offer?
> I haven’t got any offers yet on my pieces but would love to know

ht @cryptodamsky

### LINKED ISSUE #3493: Broken links in Stats to item
When stats has trailing `/`, it break item links for example. 
> from Offer Stats at BSX, I have /stats/ in the link to the item. 
> I'm curious, is this being fixed someone or reported already so I don't report it twice? Or in some ongoing PR?

image

Try click here 
- https://beta.kodadot.xyz/bsx/stats/
- and choose the item 

https://discord.com/channels/840514076538830888/899643761968173108/999275535022694421

Apparently same thing in 
- #3453

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #3450
- [x] PR closes #3493
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [x] I've tested PR on mobile and everything seems works
- [x] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

Route is `/bsx/myoffers`
https://deploy-preview-3453--koda-nuxt.netlify.app/bsx/myoffers

image

image

### CODE REVIEW

**Comment by petersopko** [comment]
hmm, I see @yangwao's initial table draft didn't include expiration time, maybe we want to include that as well?

**Comment by petersopko** [comment]
image

Can we change the message if user isn't logged in to something like "Connect wallet to see your offers here"?

**Comment by yangwao** [comment]
> didn't include expiration time

oh yes, good point!

**Comment by Jarsen136** [comment]
logged in
image


not logged in
image

**Comment by Jarsen136** [comment]
> hmm, I see @yangwao's initial table draft didn't include expiration time, maybe we want to include that as well?
> 
> such as [#3452 (review)](https://github.com/kodadot/nft-gallery/pull/3452#pullrequestreview-1041090568)

Yes, let's wait till PR #3452 is finished. So that I don't need to do the same work.

**Comment by yangwao** [comment]
this is amazing works as intended! 
I'm thinking of speeding up accepting offers. 

Let's add the button to accept offer from that screen right away? 

image

**Comment by Jarsen136** [comment]
> this is amazing works as intended! I'm thinking of speeding up accepting offers.
> 
> Let's add the button to accept offer from that screen right away?
> 

Oki, I will make it. 🚶

**Comment by yangwao** [comment]
I've just noticed that it isn't showing only my offers? 

And links to items seem broke :|

image

**Comment by Jarsen136** [comment]
> I've just noticed that it isn't showing only my offers but for whole marketplace? 
> 
> 
> 
> And links to items seem broke :|
> 
> 
> 

It should have only show one's offers. I will check them later.

**Comment by yangwao** [comment]
> It should have only show one's offers. I will check them later.

now I checking again, I was just misled by my address showing there, so incoming offers seems ok 😄

offers created by user should be probably then done here hmm https://github.com/kodadot/nft-gallery/issues/3203

**Comment by petersopko** [comment]
> Yes, let's wait till PR #3452 is finished. So that I don't need to do the same work.

it's merged now :) just one more thing, I just realized we don't have route for this anywhere on the page? let's add it above Transfer in navbar for now
image

**Comment by Jarsen136** [comment]
Tested accepting several offers and they work well.

image


image

**Comment by yangwao** [comment]
> image

wait, you are logged as `bXm...` and you see an incoming offer from that address? I guess this query should be only for offers you can accept.

Offers user made being done in #3477 

We can later merge in some #3463 once it is done quite well :)

**Comment by Jarsen136** [comment]
> wait, you are logged as bXm... and you see an incoming offer from that address? I guess this query should be only for > offers you can accept.

Because I make an offer for an nft first and then I buy the same nft.
These offers all belong to my own nfts.

image

I could remove the offer that was made by the nft owner, If you feel it's strange. @yangwao

**Comment by JustLuuuu** [comment]
Links in a first column of the table (NFT) is leading nowhere.
image

**Comment by Jarsen136** [comment]
> I guess this query should be only for > offers you can accept.

✅  done

image

**Comment by petersopko** [comment]
> I could not reproduce this problem. Could you pls try again? 

it happens for me as well :) seems that if you click on the item it goes to /bsx/myoffers/gallery/{id} as opposed to desired /bsx/gallery/{id}

**Comment by Jarsen136** [comment]
> > I could not reproduce this problem. Could you pls try again?
> 
> it happens for me as well :) seems that if you click on the item it goes to /bsx/myoffers/gallery/{id} as opposed to desired /bsx/gallery/{id}

✅ Fixed

The bug could be reproduced only when the url end of '/myoffers' instead of `/myoffers/`

**Comment by roiLeo** [comment]
Is it possible to change page path name? 
it's weird to name it 'myoffer' since we don't have 'myprofile' or 'mynfts' and it would be nice to have something consistent.

**Comment by Jarsen136** [comment]
> > > `/incomingoffers`
> > 
> > 
> > ✅ rename page path
> 
> Probably rename it in the profile to "Incoming Offers" and include it in translations as well?

✅  rename title done

image

File: utils/mixins/offerMixin.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  public currentBlock = 0
+
+  created() {
+    onApiConnect(async (api) => {
+      const currentBlock = await api.query.system.number()
+      this.currentBlock = currentBlock.toNumber()
+    })
+  }
+
+  public calcExpirationTime(expirationBlock: number) {
```
Comment (kkukelka) [inline utils/mixins/offerMixin.ts:31]:
```suggestion
  public calcExpirationTime(expirationBlock: number): string {
```

File: components/Navbar.vue [hunk 2]
```diff
@@ -168,6 +168,10 @@ export default class NavbarMenu extends mixins(PrefixMixin) {
     return this.urlPrefix === 'rmrk' || this.urlPrefix === 'westend'
   }
 
+  get isBsx(): boolean {
```
Comment (kkukelka) [inline components/Navbar.vue:171]: above in the code there is a check `   v-if="urlPrefix === 'bsx'"`
could you replace this with `isBsx`

**Review by kkukelka — CHANGES_REQUESTED** [review]
2 minor thingies

File: components/Navbar.vue (hunk 2 above)
Comment (Jarsen136) [reply components/Navbar.vue:171]: oki

**Comment by JustLuuuu** [comment]
wfm

**Review by kkukelka — APPROVED** [review]
lg now

**Comment by kkukelka** [comment]
pay 300

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $300 @ 59.6 USD/KSM ~ 5.034 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0x005583760c52021dcc65d41c4a74807a2f648e45898ff051cbf98e4eb89299cd](https://kusama.subscan.io/extrinsic/0x005583760c52021dcc65d41c4a74807a2f648e45898ff051cbf98e4eb89299cd)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by yangwao** [comment]
pay 300

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $300 @ 59.37 USD/KSM ~ 5.053 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0x8740f2220cc4c4f458aef7cb7fc039454184bbd53970d9693826bad9128b1760](https://kusama.subscan.io/extrinsic/0x8740f2220cc4c4f458aef7cb7fc039454184bbd53970d9693826bad9128b1760)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**